routes/ecom/modules/apply-discount.js

```diff
@@ -163,7 +163,7 @@ module.exports = appSdk => {
             let value = 0
             rule.product_ids.forEach(productId => {
               const item = params.items.find(item => productId === item.product_id)
-              if (item) {
+              if (item && item.flags && item.flags.includes('freebie')) {
```

Comment:
Isso é um mock meio feio/errado na verdade 😬 

Em qual situação o preço do produto que é dado como brinde deveria ser considerado para calcular o montante para aplicar desconto por brinde? Isso não faz sentido pra mim ainda.
O subtotal a checar com o montante mínimo tem que excluir o preço do brinde, mesmo que esse produto não tenha sido considerado brinde ainda, pra poder aplicar o desconto e ainda assim o subtotal bater com o mínimo...

Reply:
Bom, na situação em que o produto não é um brinde ainda. Pensa com a cabeça do cliente. A tia sonia coloca brinde a partir de 200 reais. Sendo que nesse brinde é uma granola. No pedido, o cliente com certeza vai levar uma granola também, porque é o produto que é mais vendido. E ai o cara coloca 5 granolas no carrinho bate o valor minimo, ficando 201 reais e não ganha nenhum desconto, ou ele vai reclamar, ou ele vai fechar o pedido achando que o brinde entrará depois. O cliente, esperaria que quando recebesse o pedido, teria direito a 6 granolas e nao 5, porque uma teria que vir como brinde. 

No código atual, ele retira da conta, ou seja, pro cara comprar 5 e levar 6 granolas, nao funciona. Ele precisa colocar por exemplo 240 no carrinho, ao invés de 200, pra conseguir levar o brinde, o erro está ai. Se você faz uma promo, que o mínimo é 200, mas o cliente precisa colocar 240, está errado. Isso, um dos itens do carrinho for vir a ser um brinde. Se não for, de fato, acima dos 200 reais vai dar ok. 

Outra coisa que reclamam muito é por exemplo, para calculo de desconto, contabilizar o valor dos brindes. No caso, se fosse retirar, deveria ser em todos os tipos de desconto e o brinde vir como um outro id de produto no carrinho, com referência product_id, conforme vem no produto com customização.

Enfim, a se pensar